script.js

```diff
@@ -0,0 +1,96 @@
+const courseContainer = document.querySelector(".courses-container");
+const searchButton = document.querySelector(".search-btn");
+let courses = [];
+searchButton.addEventListener("click", function(e) {
+    e.preventDefault();
+    updateTable();
+});
+const searchText = document.querySelector(".search-input");
+searchText.addEventListener("keyup", function(e) {
+    updateTable();
+});
+
+function updateTable() {
+    let here = [];
+    here = filter(searchText.value);
+    courseContainer.innerHTML = "";
+    for (let i = 0; i < courses.length; i++) {
+        render(here[i]);
+    }
+}
+
+function filter(searchText) {
+    let filtered = [];
+    if (searchText == "") {
+        return courses;
+    }
+    for (let i = 0; i < courses.length; i++) {
+        if (courses[i]['title'].toLowerCase().indexOf(searchText.toLowerCase()) !== -1) {
+            filtered.push(courses[i]);
+        }
+    }
+    return filtered;
+}
+
+fetch('https://api.npoint.io/74d742bda6f905ced3c7')
+    .then((response) => {
+        return response.json();
+    }).then((arr) => {
+        courses = arr;
+        for (let i = 0; i < arr.length; i++) {
+            render(arr[i]);
+        }
+    });
```

Comment:
you should use your json-server here,  right?

Reply:
yes json-server